cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go

```diff
@@ -253,10 +253,43 @@ func (ng *nodegroup) Nodes() ([]cloudprovider.Instance, error) {
 	// must match the ID on the Node object itself.
 	// https://github.com/kubernetes/autoscaler/blob/a973259f1852303ba38a3a61eeee8489cf4e1b13/cluster-autoscaler/clusterstate/clusterstate.go#L967-L985
 	instances := make([]cloudprovider.Instance, len(providerIDs))
-	for i := range providerIDs {
-		instances[i] = cloudprovider.Instance{
-			Id: providerIDs[i],
+	for i, providerID := range providerIDs {
+		providerIDNormalized := normalizedProviderID(providerID)
+		instance := cloudprovider.Instance{
+			Id: providerID,
 		}
+
+		// Add instance Status to report instance state to cluster-autoscaler.
+		// Failed instances are reported as creating with an error,
+		// pending instances are reported as creating,
+		// and deleting instances are reported as deleting.
+		// This helps cluster-autoscaler make better scaling decisions.
+		switch {
+		case isFailedMachineProviderID(providerIDNormalized):
+			klog.V(4).Infof("Machine failed in %s (%s)", ng.Id(), providerID)
```

Comment:
nit: can we say `Machine failed in node group %s (%s)`...

for all 3 log instances here